app/code/Magento/Catalog/Model/ResourceModel/Attribute.php

```diff
@@ -80,6 +80,23 @@ protected function _afterSave(\Magento\Framework\Model\AbstractModel $object)
         return parent::_afterSave($object);
     }
 
+    /**
+     * Perform actions before object delete
+     *
+     * @param \Magento\Framework\Model\AbstractModel $object
+     * @return $this
+     */
+    protected function _beforeDelete(\Magento\Framework\Model\AbstractModel $object)
```

Comment:
Hi @mohitp196 thanks for the pull request! 

I think this logic should be moved to the repository and it's better to implement the validation in the same way as it's done in `\Magento\Eav\Model\AttributeManagement`

Reply:
@sivaschenko Please correct me if I am wrong, As per my understanding you would want me to move the code check to below file:
Magento\Catalog\Model\Product\Attribute\Repository delete() function

Reply:
Yes, I think it will be a better place for such validation

Reply:
@sivaschenko I have checked with the Attribute `\Magento\Eav\Model\AttributeRepository::delete` but I couldn't find the relevant API call for that in webapi.xml for EAV module. Instead I found code in `Magento\Eav\Model\AttributeManagement::unassign` which throws the similar exception. referring that I have changed code in the `Magento\Catalog\Model\Product\Attribute\Repository::delete` Could you please review the same and let me know if any further more changes are required.